test/replication-luatest/gh_10087_update_quorum_in_on_commit_trigger_test.lua

```diff
@@ -0,0 +1,92 @@
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_test('test_one_member_cluster_insertion', function(cg)
+    cg.master = server:new{alias = 'master'}
+    cg.master:start()
+    -- Make `_cluster` space synchronous.
+    cg.master:exec(function()
+        box.ctl.promote()
+        box.space._cluster:alter{is_sync = true}
+    end)
+end)
+
+g.after_test('test_one_member_cluster_insertion', function(cg)
+    cg.master:drop()
+    if cg.replica ~= nil then
+        cg.replica:drop()
+    end
+end)
+
+-- Test that synchronous insertion into 1-member cluster works properly.
+g.test_one_member_cluster_insertion = function(cg)
+    cg.replica = server:new{alias = 'replica', box_cfg = {
+        replication = cg.master.net_box_uri,
+    }}
+    cg.replica:start()
+    cg.master:wait_for_downstream_to(cg.replica)
+    cg.replica:exec(function()
+        t.assert_not_equals(box.space._cluster:get{box.info.id}, nil)
+    end)
+end
+
+g.before_test('test_three_member_cluster_insertion', function(cg)
+    cg.replica_set = replica_set:new{}
+    cg.master = cg.replica_set:build_and_add_server{alias = 'master'}
+    cg.master:start()
+    cg.replica_to_be_disabled =
+        cg.replica_set:build_and_add_server{alias = 'to_be_disabled',
+                                            box_cfg = {
+        replication = {
+            cg.master.net_box_uri,
+            server.build_listen_uri('replica', cg.replica_set.id),
+        },
+    }}
+    cg.replica = cg.replica_set:build_and_add_server{alias = 'replica',
+                                                     box_cfg = {
+        replication = {
+            cg.master.net_box_uri,
+            server.build_listen_uri('to_be_disabled', cg.replica_set.id),
+        },
+    }}
+    cg.replica_set:start()
+
+    -- Make `_cluster` space synchronous.
+    cg.master:exec(function()
+        box.ctl.promote()
+        box.space._cluster:alter{is_sync = true}
+    end)
+
+    cg.master:wait_for_downstream_to(cg.replica_to_be_disabled)
+    cg.master:wait_for_downstream_to(cg.replica)
+end)
+
+g.after_test('test_three_member_cluster_insertion', function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Test that synchronous insertion into 3-member cluster with 1 disabled node
+-- works properly.
+g.test_three_member_cluster_insertion = function(cg)
```

Comment:
Could you please move the `after_test` after the test? Otherwise it really complicates the reading - I have to jump back above the test's code to see what it does in the end.

Reply:
And we can also make different test groups just for the sake of code readability. Tests seems to be really independent.

Reply:
> Could you please move the `after_test` after the test?

> And we can also make different test groups just for the sake of code readability. Tests seems to be really independent.

Tests groups consisting of 1 tests sounds like overengineering, I don't think it will really improve the code readibility — let's leave it as is.